episodes/07-infrastructure.md

```diff
@@ -29,18 +21,329 @@ exercises: 15
 
 ::::::::::::::::::::::::::::::::::::::::::::::::::
 
+At this stage in the training, 
+you should have a clear idea of who the people are that you want to teach this lesson to,
+and what exactly the skills are that you want them to learn while they are following it.
+It is time to begin creating a website that will present that lesson to the world!
+
+## GitHub Pages
+
+The source of all The Carpentries lessons is made publicly-available in repositories on [GitHub].
+By making our repositories public like this,
+we encourage others to help us maintain and improve our lessons,
+and make it as easy as possible for them to re-use and modify our lessons for their own purposes.
+
+GitHub provides a hosting service to open source projects such as these,
+allowing users to present their projects to the wider world.
+The repository includes a complete history of the changes made 
+between versions of the individual files in the project,
+and provides many features that facilitate collaboration on projects.
+We will learn more about some of those collaborative features later in this training.
+For now, we will focus on one other important feature that GitHub provides: website hosting.
+
+Via a system called **GitHub Pages**, users can build and host websites 
+from the files present in any repository on GitHub.
+For many years, 
+this has been how The Carpentries presents its lesson websites to the world.
+In the past, our lesson sites were generated by software called _[Jekyll]_, 
+a tool built into GitHub Pages that allows the webpage content, 
+written in the text files of the repository, 
+to be combined with descriptions of settings, 
+structure,
+and styling,
+to create a relatively complex website.
+
+## Using The Carpentries Workbench
+
+In 2022, we adopted a new infrastructure for our lesson sites: **The Carpentries Workbench**.
+Lesson sites built on the Workbench are still hosted with GitHub Pages,
+but no longer use Jekyll.
+Instead, the lessons are built using a programming language, _[R]_, and _[pandoc]_, 
+a software designed for converting content between file formats.
+The Workbench combines three R packages:
+
+- `{sandpaper}`: 
+   converts a collection of Markdown or RMarkdown files into
+   the structure of a lesson website.
+- `{varnish}`: provides the files and folders that add 
+   styling and additional functionality to a lesson website.
+- `{pegboard}`: an programmatic interface to the lesson, 
+   enabling various automated validation tasks.
+
+For lesson developers, 
+we believe that the Workbench makes The Carpentries lesson repositories
+much simpler to navigate and work with.
+We will use it now to initialise a new lesson.
```

Comment:
@zkamvar Questions related to the block above: Why/how are `{sandpaper}` and `{pegboard}` converted into links in the built page? Furthermore, why _isn't_ `{varnish}`?

Reply:
I have reorganised in eadb31a. I used the opportunity to expand the history a little bit, to better credit some of the original work.

Reply:
> @zkamvar Questions related to the block above: Why/how are `{sandpaper}` and `{pegboard}` converted into links in the built page? Furthermore, why _isn't_ `{varnish}`?

This is partially because of the {downlit} package, which is used by {pkgdown} (and thus {sandpaper}) to auto-add links to external documentation. Both {sandpaper} and {pegboard} are linked because I've built documentation sites for them, but I haven't done so for {varnish} because it is a package in name only and just holds css/js/html templates.